src/what-unsafe-does.md

```diff
@@ -7,6 +7,8 @@ The only things that are different in Unsafe Rust are that you can:
 * Implement `unsafe` traits
 * Mutate statics
 * Access fields of `union`s
+* Cast raw pointers to integers within const contexts
+* Compare raw pointers within const contexts
```

Comment:
With the `guaranteed_*` intrinsics, we could actually disallow the latter entirely, even in unsafe code, right?